sql/stats/join.go

```diff
@@ -0,0 +1,555 @@
+package stats
+
+import (
+	"container/heap"
+	"fmt"
+	"log"
+	"math"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+var ErrJoinStringStatistics = errors.New("joining string histograms is unsupported")
+
+// Join performs an alignment algorithm on two sets of statistics, and then
+// pairwise estimates bucket cardinalities by joining MCVs directly and
```

Comment:
I would avoid abbreviating MCV in these docstrings. For someone who doesn't know what it stands for (ie, me) it makes it easier for them to figure out what the function is doing.